assets/app/scripts/services/logs/logLinks.js

```diff
@@ -0,0 +1,71 @@
+'use strict';
+
+angular.module('openshiftConsole')
+  .factory('logLinks', [
+    '$anchorScroll',
+    '$document',
+    '$location',
+    '$window',
+    function($anchorScroll, $document, $location, $window) {
+      var doc = $document[0];
+      return {
+        makeDownload: function(obj) {
+          var a = doc.createElement('a');
+          a.href = $window.URL.createObjectURL(new Blob([obj], {type: 'text/plain'}));
+          a.download = 'log.txt';
+          doc.body.appendChild(a);
+          a.click();
+          doc.body.removeChild(a);
+        },
```

Comment:
Ok I wasn't sure if the url object needed to exist until the user is
finished "downloading" the content. But if revoking at that point works
then I agree its the right place to do it.

On Mon, Aug 24, 2015 at 2:25 PM, Ben Petersen <notifications@github.com>
wrote:

> In assets/app/scripts/services/logs/logLinks.js
> <https://github.com/openshift/origin/pull/4322#discussion_r37784518>:
>
> > +  .factory('logLinks', [
> > +    '$anchorScroll',
> > +    '$document',
> > +    '$location',
> > +    '$window',
> > +    function($anchorScroll, $document, $location, $window) {
> > +      var doc = $document[0];
> > +      return {
> > +        makeDownload: function(obj) {
> > +          var a = doc.createElement('a');
> > +          a.href = $window.URL.createObjectURL(new Blob([obj], {type: 'text/plain'}));
> > +          a.download = 'log.txt';
> > +          doc.body.appendChild(a);
> > +          a.click();
> > +          doc.body.removeChild(a);
> > +        },
>
> Yeah, I'm inclined to URL.revokeObjectURL(objectURL) immediately after
> the doc.body.removeChild(a);. The user will already have the file & we
> don't have to worry about memory. I can't see a user clicking the link
> enough times to cause a performance issue, even if they ask for the same
> log more than once. Its probably the cleanest way to ensure nothing lingers.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/openshift/origin/pull/4322/files#r37784518>.
>